primitives/src/asset.rs

```diff
@@ -160,6 +163,8 @@ pub struct GranularCanTransferResult {
     pub transfer_condition_result: Vec<TransferConditionResult>,
     /// Result of compliance check.
     pub compliance_result: AssetComplianceResult,
+    /// The weight for checking the asset's compliance and transfer restrictions.
+    pub consumed_weight: Option<Weight>,
     /// Final evaluation result.
     pub result: bool,
 }
```

Comment:
This is a breaking change to the Runtime APIs (RPC -> runtime interface) and requires bumping the API version in https://github.com/PolymeshAssociation/Polymesh/blob/8a502aea8f8e47953c222a2a6da575282f683ee4/rpc/runtime-api/src/asset.rs#L31